Bottom: 2fab5dd88e216f1c46e2e38d7a17d74218db5b12
Top:    7b3f2f72d6d89dc38bcafed49d419c43e222d827
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-09-09 13:12:51 +0100

Refresh of fix-layer-on-off-rendering

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index bf0c723..f8e4426 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1379,7 +1379,7 @@ DrawLayerGroup (int group, const BoxType * screen)
         strcmp (Layer->Name, "route") == 0)
       rv = 0;
 
-    if (layernum < max_layer /*&& Layer->On*/) {
+    if (layernum < max_layer && Layer->On) {
 
       if (!first_run)
         gui->set_layer (0, group, 0);
@@ -1540,7 +1540,7 @@ ghid_draw_everything (BoxTypePtr drawn_area)
     l = global_view_2d ? LAYER_ON_STACK (i) : LAYER_PTR (orderi);
     group = GetLayerGroupNumberByNumber (global_view_2d ? LayerStack[i] : orderi);
 
-    if (l->On && !do_group[group]) {
+    if (!do_group[group]) {
       do_group[group] = 1;
       drawn_groups[ngroups++] = group;
     }
